engine/src/main/java/org/terasology/engine/rendering/nui/asset/UIFormat.java

```diff
@@ -175,7 +175,7 @@ public UIWidget deserialize(JsonElement json, Type typeOfT, JsonDeserializationC
             if (id != null) {
                 FieldMetadata<?, ?> fieldMetadata = elementMetadata.getField(ID_FIELD);
                 if (fieldMetadata == null) {
-                    logger.warn("UIWidget type {} lacks id field", elementMetadata.getId());
+                    logger.atWarn().addArgument(() -> elementMetadata.getId()).log("UIWidget type {} lacks id field");
```

Comment:
Simple getter for member access.
```suggestion
                    logger.warn("UIWidget type {} lacks id field", elementMetadata.getId()); //NOPMD
```